test/pytest/test_xpedite/test_profiler/generateBaseline.py

```diff
@@ -9,39 +9,103 @@
 """
 
 import os
-from shutil                               import copy, rmtree
+import json
+import cPickle as pickle
 import logging
 import logging.config
+import argparse
+from shutil                               import copy, rmtree
 from logger                               import LOG_CONFIG_PATH
-from test_xpedite.test_profiler.profile   import (
-                                            buildNotebook, loadProfileInfo, loadProbes,
+from xpedite.util                         import makeLogPath
+from xpedite.transport.remote             import Remote
+from xpedite.transport.net                import isIpLocal
+from xpedite.benchmark                    import makeBenchmark
+from test_xpedite.test_profiler.profile   import buildNotebook, loadProbes
+from test_xpedite                         import (
+                                            REPORT_CMD_BASELINE_PATH,
+                                            BASELINE_CPU_INFO_PATH,
+                                            PROBE_CMD_BASELINE_PATH, DATA_FILE_EXT,
+                                            XPEDITE_APP_INFO_PATH, TXN_COUNT,
+                                            THREAD_COUNT
                                           )
+from test_xpedite.test_profiler.app       import TargetLauncher
+from test_xpedite.test_profiler.profile   import generateProfiles
+from test_xpedite.test_profiler.context   import Context
+from test_xpedite.test_profiler.scenario  import ScenarioLoader
 
 logging.config.fileConfig(LOG_CONFIG_PATH)
 LOGGER = logging.getLogger('xpedite')
+BENCHMARK_SCENARIO = 'benchmark'
+ARCHIVE_EXT = '*.tar.gz'
+BENCHMARK_APP_INFO_PATH = os.path.join(
+  BENCHMARK_SCENARIO, BENCHMARK_SCENARIO, 'appinfo.txt'
+)
+DATA_FOLDER = os.path.join(os.path.dirname(__file__), '..', 'data')
+APPS = ['allocatorApp', 'dataTxnApp', 'multiThreadedApp', 'slowFixDecoderApp']
+DATA_FILE_EXT_REGEX = r'*{}'.format(DATA_FILE_EXT)
+BENCHMARK_REGEX = r'/{}'.format(BENCHMARK_SCENARIO)
+
+class BaselineGenerator(object):
+  """
+  Set up and environment for loading scenarios in a remote or local environment
+  """
+
+  def __init__(self, runDir, hostname, context):
+    self.remote = None
+    self.runDir = runDir
+    self.hostname = hostname
+    self.context = context
+    if not isIpLocal(hostname):
+      self.remote = Remote(hostname, makeLogPath('remote'))
+    self.scenarioLoader = ScenarioLoader()
+
+  def __enter__(self):
+    if self.remote:
+      self.remote.__enter__()
+    cleanUpDataDir(self.runDir)
+    self.scenarioLoader.loadScenarios(self.runDir, APPS, remote=self.remote)
+    self.benchmarkProfile()
+    return self
+
+
+  def __exit__(self, excType, excVal, excTb):
+    if self.remote:
+      self.remote.__exit__(None, None, None)
+
+  def benchmarkProfile(self):
```

Comment:
I think that this might cause a problem with validating benchmarks in scenario.py, because the benchmarks need to get generated after scenario.__enter__(), so validating them on loading will cause problems with generateBaseline.py

Reply:
^ works for
if BENCHMARK_SCENARIO not in scenario.name:
Running for the regular scenarios and creating benchmarks before benchmark scenarios load